lgc/builder/BuilderImpl.cpp

```diff
@@ -416,19 +419,16 @@ class TraceNonUniformIndex {
   const DenseMap<Value *, TinyInstructionSet> &getInstrDeps() const { return instrDeps; }
 
   const TinyInstructionSet::IndexToInstructionVec &getIndexToInstr() const { return indexToInstr; }
-
-  bool foundDependencies() const { return scalarizeDescriptorLoads; }
 };
 
 // Adds newValue in instrDeps map. The dependencies of the newValue are the currentVisitedInstr and its dependencies.
 // @param newValue : the new value to be added in instrDeps map
 // @param currentVisitedInstr : the value from where we copy the dependencies for newValue
 void TraceNonUniformIndex::insertNewValueInInstrDeps(Value *newValue, Instruction *currentVisitedInstr) {
-  if (!instrToIndex.contains(currentVisitedInstr)) {
+  if (!instrToIndex.contains(currentVisitedInstr))
     // The instruction is either outside of 64 limit or in a different basic block. So, we bail-out scalarization.
-    scalarizeDescriptorLoads = false;
```

Comment:
> - scalarizeDescriptorLoads = false;

Happy to see this part from the previous commit reworked. With the updated code we should see more scalarization happening.